src/odemis/driver/test/xt_client_test.py

```diff
@@ -713,62 +713,102 @@ def test_apply_autofocus(self):
 
     def test_get_mpp_orientation(self):
         """
-        Simple test, checks if no error occur during calling the method and the return value.
+        Test getting the multiprobe pattern orientation.
         """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
         mpp_orientation = self.microscope.get_mpp_orientation()
         self.assertIsInstance(mpp_orientation, float)
         # TODO K.K. check range with Wilco
         self.assertTrue(0 < mpp_orientation < 360)
 
     def test_get_aperture_index(self):
         """
-        Simple test, checks if no error occur during calling the method and the return value.
+        Test getting the aperture index.
         """
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
         aperture_index = self.microscope.get_aperture_index()
         self.assertIsInstance(aperture_index, float)
-        self.assertTrue(0 < aperture_index < 14)
+        self.assertTrue(0 <= aperture_index <= 14)
 
     def test_set_aperture_index(self):
         """
-        Simple test, checks if no error occur during calling the method.
+        Test setting the aperture index.
         """
-        # TODO Check if setting of the value actually happens and if an int value is also allowed.
-        self.microscope.set_aperture_index(10)
+        if self.xt_type != 'xttoolkit':
+            self.skipTest("This test needs XTToolkit to run.")
+
+        range = tuple(self.microscope.aperture_index_info()["range"])
+        current_aperture_index = self.microscope.get_aperture_index()
+        new_aperture_index = current_aperture_index + 1 if range[0] <=current_aperture_index < range[1] else \
```

Comment:
missing space after <=

Reply:
I think it is enough if you only check `current_aperture_index < range[1]` as you add `+1`

Reply:
That wouldn't work if the current index is exactly the max of the current range.

Reply:
It was about removing "range[0] <=" this probably can be removed isn't it? @Kleijwegt